### DISCUSSION TITLE
Easiest Way to Modify/Add/Remove Themes?

### DISCUSSION THREAD

**Discussion by ilohmar** [neurocyte/flow#22] [Q&A] [answered]
I wanted to play around with the built-in themes, so I cloned the flow-themes repo, with the intention of building my own custom `themes.zig` to be used in flow then.  But `zig build` in flow-themes failed --- it seems to not find the first theme file `default.json`.

Now that may be an issue to file in the flow-themes repo, but I did not want to create an XY problem, hence asking here: Am I going about this the right/wrong way?  If `zig build` in flow-themes should just work, I will gladly file an issue there.

Thanks!

**Comment by neurocyte** [comment, marked as answer]
Yes, `zig build` in flow-themes should just work. This will produce a zig module in `zig-out` that can then be referenced in flow. You can use your local build of flow-themes by changing the `.url` line in flow's `build.zig.zon` to `.path = "{flow-themes-path}/zig-out"` and remove  the `.hash` line.

If you can't get flow-themes to build you are probably not using the correct version of zig. Check the version stored in `build.zig.version` and make sure it matches your zig version. Or use the `./zig` wrapper script to automatically fetch the correct version and build.

If it still doesn't build please open an issue in the flow-themes repo.

**Comment by neurocyte** [comment]
For future reference I'll mention the solution to [flow-themes#1](https://github.com/neurocyte/flow-themes/issues/1) here. To build flow-themes you need to have `hjson` installed in your path. It is used to sanitize theme files that are not strict json.
